### PR TITLE
spectrum-Search-icon should be pushed down into TextFieldBase

### PR DESCRIPTION
**Pull request by so99ynoodles**
Closes https://github.com/adobe/react-spectrum/issues/790

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Changed the className of the default icon of SearchField and deleted unused CSS.

- Check if the style of the icon of SearchField is proper.

### CODE REVIEW

**Review by devongovett — APPROVED** [review]
LGTM. @snowystinger @LFDanLu?

**Comment by LFDanLu** [comment]
From what I can remember when I worked on this way back when, the Search icon/SearchField was a bit different in the XD files than it was in the TextField w/ icon designs at the time (sizing + positioning). Haven't taken a look at if there were are any updates, lemme do that now just to make sure

**Review by LFDanLu — DISMISSED** [review]
Double checked and looks like the old SearchField icon position is to match the "Search-field-beta-v1.1.0" XD files from the spectrum contribution site. Its pretty old though (May 12, 2020) and spectrum-css uses the TextField icon positioning for their [SearchFields](https://opensource.adobe.com/spectrum-css/search.html) (albeit with different values being set than what we have currently such as `left` being 12px rather than 8px). IMO what's here is fine, gets us closer to parity with spectrum css.

File: packages/@react-spectrum/searchfield/src/SearchField.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ function SearchField(props: SpectrumSearchFieldProps, ref: RefObject<TextFieldRe
       UNSAFE_className={
         classNames(
           styles,
-          'spectrum-Search-icon'
+          'spectrum-Textfield-icon'
```
Comment (snowystinger) [inline packages/@react-spectrum/searchfield/src/SearchField.tsx:33]: I don't believe this is needed, it is not a class in `import styles from '@adobe/spectrum-css-temp/components/search/vars.css';` (or really the two files it imports)
If it is, we should name it differently

this class is added in TextFieldBase from Textfield css https://github.com/adobe/react-spectrum/blob/main/packages/%40react-spectrum/textfield/src/TextFieldBase.tsx#L92

**Review by snowystinger — COMMENTED** [review]
One minor comment, otherwise looks good, i'll test it tomorrow

File: packages/@react-spectrum/searchfield/src/SearchField.tsx (hunk 1 above)
Comment (LFDanLu) [reply packages/@react-spectrum/searchfield/src/SearchField.tsx:33]: 🤦, @snowystinger is correct here, you can go ahead and remove the entire UNSAFE_className block here

Comment (so99ynoodles) [reply packages/@react-spectrum/searchfield/src/SearchField.tsx]: Nice, thank you for checking!

**Review by devongovett — DISMISSED** [review]

**Comment by snowystinger** [comment]
With the change to the size of the icon, we need to update the base decorated textfield most likely, I've pushed those changes which we'll run past design. Thank you! and sorry for the delay

**Review by snowystinger — APPROVED** [review]
LGTM

**Review by LFDanLu — APPROVED** [review]

**Pull request merged**